### PR TITLE
Added getAllProjects() to retrieve all user's private and public projects. Minor refactorings. 

### PR DESCRIPTION
**Pull request by paulolc**
Added getAllProjects() to retrieve all private and public projects of the Scratch user and minor refactorings to remove duplicated and commented code.

### CODE REVIEW

File: scratchapi.js [hunk 1]
```diff
@@ -7,9 +7,10 @@ var fs = require('fs');
 var WebSocket = require('ws');
 
 var SERVER = 'scratch.mit.edu';
-var PROJECTS_SERVER = 'projects.scratch.mit.edu';
-var CDN_SERVER = 'cdn.scratch.mit.edu';
-var CLOUD_SERVER = 'clouddata.scratch.mit.edu';
+var PROJECTS_SERVER = 'projects.' + SERVER;
+var CDN_SERVER = 'cdn.' + SERVER ;
+var CLOUD_SERVER = 'clouddata.' + SERVER;
+var API_SERVER = 'api.' + SERVER;
```
Comment (trumank) [inline scratchapi.js]: I would prefer these to stay as fully qualified domain names. It's easier to copy and paste.

File: scratchapi.js [hunk 2]
```diff
@@ -142,54 +146,52 @@ Scratch.UserSession.prototype.verify = function(cb) {
...
+    hostname: SERVER,
+    path: '/site-api/projects/all/',
+    method: 'GET',
+    sessionId: this.sessionId
+  },cb);
+};
 Scratch.UserSession.prototype.setProject = function(projectId, payload, cb) {
   if (typeof payload !== 'string') payload = JSON.stringify(payload);
-  request({
+  requestjson({
```
Comment (trumank) [inline scratchapi.js]: rename to `requestJSON`

File: scratchapi.js (hunk 1 above)
Comment (paulolc) [reply scratchapi.js]: Ok. Done.

File: scratchapi.js (hunk 2 above)
Comment (paulolc) [reply scratchapi.js]: Ok. Done.

**Pull request merged**